openpype/plugins/publish/validate_publish_dir.py

```diff
@@ -0,0 +1,69 @@
+import pyblish.api
+from openpype.pipeline.publish import ValidateContentsOrder
+from openpype.pipeline.publish import (
+    KnownPublishError,
+    get_publish_template_name,
+)
+
+
+class ValidatePublishDir(pyblish.api.InstancePlugin):
+    """Validates if 'publishDir' is a project directory
+
+    'publishDir' is collected based on publish templates. In specific cases
+    ('source' template) source folder of items is used as a 'publishDir', this
+    validates if it is inside any project dir for the project.
+    (eg. files are not published from local folder, unaccessible for studio'
+
+    """
+
+    order = ValidateContentsOrder
+    label = "Validate publish dir"
+
+    checked_template_names = ["source"]
+    # validate instances might have interim family, needs to be mapped to final
+    family_mapping = {
+        "renderLayer": "render",
+        "renderLocal": "render"
+    }
+
+    def process(self, instance):
+
+        template_name = self._get_template_name_from_instance(instance)
+
+        if template_name not in self.checked_template_names:
+            return
+
+        original_dirname = instance.data.get("originalDirname")
+        if not original_dirname:
+            raise KnownPublishError(
+                "Instance meant for in place publishing."
+                " Its 'originalDirname' must be collected."
+                " Contact OP developer to modify collector.")
+
+        anatomy = instance.context.data["anatomy"]
+
+        success, _ = anatomy.find_root_template_from_path(original_dirname)
+        self.log.info(_)
+        if not success:
+            raise KnownPublishError(
```

Comment:
```suggestion
            raise PublishValidationError(
```
I think that in this case it should be `PublishValidationError` as this is not code related and is a direct result of artists actions